test/scripts/integration/Run-IntegrationTests.ps1

```diff
@@ -60,6 +83,16 @@ if ($env:OS -eq 'Windows_NT') {
         File = "$PSScriptRoot\..\..\mod\amxmodx\addons\amxmodx\scripting\plugin1.sma"
         Force = $true
     }
+    $expectedExitCode = 0
+    $iterations = 2
+    & $functionTestScriptBlock
+
+    "`n[amxmodx] Compile plugin via wrapper, bad plugin" | Write-Host
```

Comment:
now that we are using regex patterns as discussed in https://github.com/startersclan/Compile-SourceScript/pull/7#discussion_r468210203, failing to detect errors should no longer happen.

Reply:
That would still be an assumption. I don't know the compilers well enough to say if that regex pattern suffices in capturing all possible errors that could occur. In fact I think what's more robust is revising the regex pattern to instead account for the pattern such as:

- `sourcemod` - https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=545 instead of https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=545
- `amxmodx` - https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=887 instead of https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=885

Reply:
> That would still be an assumption. I don't know the compilers well enough to say if that regex pattern suffices in capturing all possible errors that could occur. In fact I think what's more robust is revising the regex pattern to instead account for the pattern such as:

> - `sourcemod` - https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=549 instead of https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=545

We could match several lines. E.g. regex `^Compilation aborted\.` and `^.*\d+\s+Errors?.'^.*\d+\s+Errors?\`

```
plugin1_bad.sp(283) : error 029: invalid expression, assumed zero
plugin1_bad.sp(283) : fatal error 190: too many error messages on one line

Compilation aborted.
4 Errors.
```

> amxmodx - https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=887 instead of https://dev.azure.com/startersclan/Compile-SourceScript/_build/results?buildId=319&view=logs&j=cf913be4-6704-58f5-cde6-258cc8aef7bf&t=c3dd6012-84fa-5b88-85db-ffd0f16e84df&l=885

We could match several lines. E.g. regex `^\(compile failed\)\.$` and `^.*\d+\s+Errors?\.'^.*\d+\s+Errors?\`

```
plugin2_bad.sma(6) : error 010: invalid function or declaration

1 Error.
Could not locate output file compiled/plugin2_bad.amx (compile failed).
End of compilation.
```

I'll add those to the regex patterns if its ok @joeltimothyoh

Reply:
I think those patterns would definitely do better. @leojonathanoh

Reply:
done

Reply:
Fixed in https://github.com/startersclan/Compile-SourceScript/pull/7/commits/d9751aaa570f1164bfdc362d6f5c57fcdf2cc2b4